tee/rpc/conn.go

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package rpc
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"net"
+	"net/rpc"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/perun-network/erdstall/tee"
+
+	"perun.network/go-perun/log"
+)
+
+var _ tee.Enclave = (*RemoteEnclave)(nil)
+
+// RemoteEnclave communicates with a rpc.Node's enclave.
+type RemoteEnclave struct {
+	client *rpc.Client
+}
+
+// NewRemoteEnclave communicates with a rpc.Node via conn.
+func NewRemoteEnclave(conn io.ReadWriteCloser) *RemoteEnclave {
+	return &RemoteEnclave{
+		client: rpc.NewClient(conn),
+	}
+}
+
+// Dials an enclave three times to allow three concurrent accesses to the enclave.
+func DialEnclave(address string) (*RemoteEnclave, error) {
+	conn, err := net.Dial("tcp", address)
+	if err != nil {
+		return nil, err
+	}
+	return NewRemoteEnclave(conn), nil
+}
+
+// Close closes the connection to the remote enclave.
+func (re *RemoteEnclave) Close() error {
+	return re.client.Close()
+}
+
+func (re *RemoteEnclave) Run(p tee.Parameters) error {
+	return re.client.Call("Node.Run", p, &Void{})
+}
+
+func (re *RemoteEnclave) Shutdown() {
+	if err := re.client.Call("Node.Shutdown", Void{}, &Void{}); err != nil {
+		log.Error(err)
+	}
+}
+
+func (re *RemoteEnclave) Init() (common.Address, []byte, error) {
+	var res TeeInitRes
+	return res.Addr, res.Sig, re.client.Call("Node.Init", &Void{}, &res)
```

Comment:
Sure that this works in 1 line? If it reads the values of `res.Addr` and `res.Sig` first and then performs the call, it won't.